install-2x.html

```diff
@@ -0,0 +1,276 @@
+
+
+      </p><pre>
+#!/bin/bash
+
+# this is where Cypht will be installed
+DESTINATION="/usr/local/share/cypht"
+
+# validate the destination directory
+sudo test -r $DESTINATION -a -x $DESTINATION
+if [ $? -ne 0 ]; then
+    sudo mkdir $DESTINATION
+fi
+
+# create working directory
+mkdir cypht-temp
+cd cypht-temp
+
+# grab latest code
+wget https://github.com/cypht-org/cypht/archive/master.zip
+
+# unpack the archive
+unzip master.zip
+
+# run composer
+cd cypht-master && composer install && cd ..
+
+# create a .env file
+cp cypht-master/.env.example cypht-master/.env
+
+# fix permissions and ownership
+find cypht-master -type d -print | xargs chmod 755
+find cypht-master -type f -print | xargs chmod 644
+sudo chown -R root:root cypht-master
+
+# copy to destination folder
+sudo mv cypht-master/* $DESTINATION
+
+# remove working directory
+cd ..
+sudo rm -rf cypht-temp
+
+</pre>
+      <h4>3. Configure the program</h4>
+      <p>To configure Cypht for your environment, make adjustments to the <a href="https://github.com/cypht-org/cypht/blob/master/.env.example">.env</a> file according to your preferences. The .env file serves as the primary configuration file.</p>
+
+      <p>First edit the .env file to configure Cypht for your environment. If you choose to leverage a database for authentication, sessions, or user settings, ensure to complete the relevant sections within the .env file based on the information provided in the config/app.php file.</p>
+      
+      <p>The necessary SQL commands for creating tables can be found in the config/app.php file, specifically starting from line 617 in the "DB Sessions" section. </p>
+
+      <p>Cypht needs read, and read-write access to a few directories on the server. For security reasons these should NOT be inside the web-server document root. A good place for these is under the /var/lib/ sub-directory. Here are the commands To create the required directories per the default values in the ini file (assuming your web-server software runs as the "www-data" user).</p>
+
+      <pre>sudo mkdir /var/lib/hm3
+sudo mkdir /var/lib/hm3/attachments
+sudo mkdir /var/lib/hm3/users
+sudo mkdir /var/lib/hm3/app_data
+
+sudo chown -R www-data /var/lib/hm3/
+      </pre>
+      <p>The /var/lib/hm3/users directory is only required if you are using the file-system and not a database to store user settings (user_config_type = file in the .env). You can put these directories anywhere, just make sure the values in the ini file point to the right place.
+      </p>
+      <h4>4. Generate Dynamic Run-Time Configuration</h4>
```

Comment:
Is step 4 still required? I expect no, Last time while sharing with @josaphatim, he told me that We have to be reloading after updating .enve file. Please @Shadow243 see together If I am right.

Reply:
Yes @Baraka24.

This why we will keep it:
```
Recent updates have streamlined the process, and running the command above will now specifically generate the config/dynamic.php file. This file incorporates dynamic configurations, including 'handler_modules,' 'output_modules,' and 'input_filters.' Furthermore, the command continues to optimize the configuration and combine/minimize page assets, ensuring efficient runtime operation.

Additionally, executing this command will create a sub-directory called 'site' that contains the code and page assets required within the document root. If you ever relocate the source location, you'll need to re-run the config_gen script to update the program.
```